File: servers/robot-controller-backend/gpio/gpio_mock.go
// File: /Omega-Code/servers/robot-controller-backend/gpio/gpio_mock.go

package gpio

import "log"

// MockGPIO implements the GPIO interface for testing.
type MockGPIO struct{}

func (m MockGPIO) Open() error {
    log.Println("⚡ [MOCK] GPIO Opened")
    return nil
}

func (m MockGPIO) Close() error {
    log.Println("⚡ [MOCK] GPIO Closed")
    return nil
}

func (m MockGPIO) Pin(pin int) GPIOPin {
    return &MockGPIOPin{pin: pin, state: false} // ✅ Default state set to false (LOW)
}

// MockGPIOPin simulates GPIO pin behavior.
type MockGPIOPin struct {
    pin   int
    state bool // ✅ Changed from `rpio.State` to `bool`
}

func (p *MockGPIOPin) Input() {}

func (p *MockGPIOPin) Output() {}

func (p *MockGPIOPin) Read() bool { // ✅ Changed return type to `bool`
    return p.state
}

func (p *MockGPIOPin) High() {
    log.Printf("⚡ [MOCK] Pin %d set to HIGH\n", p.pin)
    p.state = true // ✅ High is now `true`
}

func (p *MockGPIOPin) Low() {
    log.Printf("⚡ [MOCK] Pin %d set to LOW\n", p.pin)
    p.state = false // ✅ Low is now `false`
}

func (p *MockGPIOPin) SetState(state bool) { // ✅ Added `SetState()` implementation
    p.state = state
    log.Printf("⚡ [MOCK] Pin %d set to state %v\n", p.pin, state)
}

File: servers/robot-controller-backend/gpio/gpio_mock.py
# File: /Omega-Code/servers/robot-controller-backend/gpio/gpio_mock.py

"""
This script simulates GPIO actions for testing purposes.
It provides a mock implementation of GPIO operations without requiring actual hardware.
"""

class GPIO:
    BCM = "BCM"
    OUT = "OUT"
    HIGH = True
    LOW = False

    @staticmethod
    def setmode(mode):
        print(f"GPIO setmode({mode})")

    @staticmethod
    def setup(pin, mode):
        print(f"GPIO setup({pin}, {mode})")

    @staticmethod
    def output(pin, state):
        print(f"GPIO output({pin}, {state})")

    @staticmethod
    def cleanup():
        print("GPIO cleanup()")

# Simulating the GPIO actions
def main():
    import sys
    pin = int(sys.argv[1])
    state = sys.argv[2].upper()

    GPIO.setmode(GPIO.BCM)
    GPIO.setup(pin, GPIO.OUT)

    if state == "HIGH":
        GPIO.output(pin, GPIO.HIGH)
    else:
        GPIO.output(pin, GPIO.LOW)

    GPIO.cleanup()

if __name__ == "__main__":
    main()
